File: KysectAcademyTask.Tests/FilesForTests/FiltersTests/RootDirectory/M3237/Robin Robertson/7. Банки/20191228143729/DebitAccount.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

namespace OOPLab7.Accounts
{
    class DebitAccount : Account
    {
        public DebitAccount(Client client, decimal sum = default) : base(client, sum)
        {
        }        

        public override Transaction Transfer(Account account, decimal sum, bool ignore = false)
        {
            if (ignore)
            {
                return base.Transfer(account, sum);
            }
            if (Balance < sum)
            {
                OnUpdate($"Not enough money to transfer {sum:C} from account {AccountId}");
                return null;
            }
            else
            {
                return base.Transfer(account, sum);
            }

        }

        public override Transaction Withdraw(decimal sum, bool ignore = false)
        {
            if (ignore)
            {
                return base.Withdraw(sum);
            }
            if (Balance >= sum)
            {
                return base.Withdraw(sum);
            }
            else
            {
                OnUpdate($"Not enough money to withdraw {sum:C} from account {AccountId}");
                return null;
            }
        }
        
    }
}

File: KysectAcademyTask.Tests/FilesForTests/DbTests/RootDirectory/M3236/Betty Padilla/7. Банки/20191224104913/DebitAccount.java
package entity.account;

import entity.AccountID;
import entity.Client;

public class DebitAccount extends Account
{
	private final float irate_debit;
	private float deferred = 0;

	public DebitAccount(AccountID ID, Client client, float irate_debit) {
		super(ID, client);
		this.irate_debit = irate_debit;
	}

	@Override
	public boolean withdraw(float val)
	{
		if (money >= val) {
			money -= val;
			return true;
		}
		return false;
	}

	@Override
	public void update() {
		super.update();

		deferred += money * irate_debit;

		if (daysActive % 30 == 0) {
			money += deferred;
			deferred = 0;
		}
	}
}
